### PR TITLE
checks client's authority to manage status repos during initialization

### PR DESCRIPTION
**Pull request by kezike**
This PR addresses Issue #16 by checking a client's authority to manage status repos during initialization and reporting an error message to the caller if not.

### CODE REVIEW

File: src/credential-status-manager-index.ts [hunk 1]
```diff
@@ -94,12 +94,17 @@ export async function createStatusManager(options: CredentialStatusManagerOption
...
-  const reposExist = await statusManager.statusReposExist();
-  const reposEmpty = await statusManager.statusReposEmpty();
+  const hasAccess = await statusManager.hasStatusAuthority(repoAccessToken, metaRepoAccessToken);
+  if (!hasAccess) {
+    throw new Error(`One or more of the access tokens you are using for the credential status repo ("${repoName}") and the credential status metadata repo ("${metaRepoName}") are incorrect or expired.`);
+  }
 
+  const reposExist = await statusManager.statusReposExist();
   if (!reposExist) {
     throw new Error(`The credential status repo ("${repoName}") and the credential status metadata repo ("${metaRepoName}") must be manually created in advance.`);
```
[the remark this answers is not in the document]
Comment (kezike) [reply src/credential-status-manager-index.ts:104]: OK, so kinda like what I did for [this error](https://github.com/digitalcredentials/credential-status-manager-git/blob/improve-misleading-access-token-error-message/src/errors.ts), yeah?

[the remark this answers is not in the document]
Comment (jchartrand) [reply src/credential-status-manager-index.ts:104]: Yes.  

Looking at that error, though, I'm curious about these two lines:

https://github.com/digitalcredentials/credential-status-manager-git/blob/5bd432cf79cffd89d5c44b4dfb50fe20a515f071/src/errors.ts#L22-L23

i.e, 

```
// Set the prototype explicitly.
Object.setPrototypeOf(this, StatusRepoInconsistencyError.prototype);
```

Why do you have to explicitly set the prototype?  Wouldn't the new object's prototype already point at StatusRepoInconsistencyError.prototype?

[the remark this answers is not in the document]
Comment (kezike) [reply src/credential-status-manager-index.ts:104]: OK, I added a number of new errors to `errors.ts`, including `InvalidTokenError` and `MissingRepositoryError`, as recommended. Also, I don't recall why I added that call to `Object.setPrototypeOf`. I have removed it for now.

[the remark this answers is not in the document]
Comment (kezike) [reply src/credential-status-manager-index.ts]: Sorry, I just realized that I pushed those changes to an older branch that was already merged into this one. I just merged those updates into this branch.

**Review by jchartrand — APPROVED** [review]
Lovely new Error classes!

**Pull request merged**